### PR TITLE
adds minimal verification to ironfish transactions for evm

### PR DESCRIPTION
**Pull request by jowparks**
## Summary
Simple `EvmDescription` verification via the EVM. This is a roundtrip transaction that passes through rust, to node typescript, and then to EVM.

Simply verifies the transaction is executable. Only form of validation explicitly done is signature for the unit test.
## Testing Plan
tests pass
## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and label it with `breaking-change-rpc` or `breaking-change-sdk`.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/primitives/evmDescription.ts [hunk 1]
```diff
@@ -11,3 +13,30 @@ export interface EvmDescription {
...
+export function evmDescriptionToLegacyTransaction(desc: EvmDescription): LegacyTransaction {
+  return new LegacyTransaction({
+    nonce: desc.nonce,
+    to: desc.to.length > 0 ? desc.to : undefined,
+    value: desc.value,
+    data: desc.data,
+    v: BigInt(desc.v),
+    r: desc.r.length > 0 ? bytesToBigInt(desc.r) : undefined,
+    s: desc.s.length > 0 ? bytesToBigInt(desc.s) : undefined,
+    // TODO(jwp) gas constants are hardcoded
```
Comment (jowparks) [inline ironfish/src/primitives/evmDescription.ts:38]: I have this hardcoded, we could potentially just add to the description for now, wdyt?

File: ironfish/src/wallet/wallet.ts [hunk 2]
```diff
@@ -98,7 +99,7 @@ export class Wallet {
   readonly scanner: WalletScanner
   readonly nodeClient: RpcClient | null
   private readonly config: Config
```
Comment (jowparks) [inline ironfish/src/wallet/wallet.ts:101]: Needs to be readonly to mock.

File: ironfish-cli/src/commands/test-evm.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import { LegacyTransaction } from '@ethereumjs/tx'
 import { Account, Address } from '@ethereumjs/util'
 import { generateKey } from '@ironfish/rust-nodejs'
```
Comment (jowparks) [inline ironfish-cli/src/commands/test-evm.ts:6]: This import was failing for me, not sure why

**Review by hughy — APPROVED** [review]

File: ironfish/src/primitives/evmDescription.ts (hunk 1 above)
Comment (jowparks) [reply ironfish/src/primitives/evmDescription.ts:38]: Is there a way to generically disable gas int eh EVM?  If so we should do it.

File: ironfish/src/consensus/verifier.ts [hunk 4]
```diff
@@ -548,6 +562,36 @@ export class Verifier {
...
+      return { valid: true }
+    }
+    if (!this.chain.evm) {
+      return { valid: false, reason: VerificationResultReason.EVM_NOT_INITIALIZED }
+    }
+    // verify signature
+
+    const tx = evmDescriptionToLegacyTransaction(evmDescription)
+    try {
+      const result = await this.chain.evm.runTx({ tx })
```
[the remark this answers is not in the document]
Comment (jowparks) [reply ironfish/src/consensus/verifier.ts:578]: Good call, whatever db transaction passthrough mechanism we add we can revert in this method. I will add a TODO in here.

File: ironfish/src/blockchain/blockchain.ts [hunk 5]
```diff
@@ -61,6 +62,8 @@ export class Blockchain {
   location: string
   files: FileSystem
   consensus: Consensus
+  // TODO: the EVM creation is async, so this has to be optional for the constructor
+  evm?: IronfishEvm
```
[the remark this answers is not in the document]
Comment (jowparks) [reply ironfish/src/blockchain/blockchain.ts:66]: Is it possible to make in non-async with this library? I don't see a standard method for it in the lib.

**Pull request merged**